### PR TITLE
Move location engine to options

### PR DESCRIPTION
**Pull request by kmadsen**
## Description

Breaking this large change into smaller pieces: https://github.com/mapbox/mapbox-navigation-android/pull/3191

This one will move the LocationEngine to one place, and is a minimal change to remove the LocationEngine from the MapboxNavigation constructor

An in parallel change, is to add some functionality to core that allows us to receive location updates before the session has been started https://github.com/mapbox/mapbox-navigation-android/pull/3240.

This pull request has no functional change.

- [ ] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [ ] I have added the appropriate milestone and project boards

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR
- [ ] We might need to update / push `api/current.txt` files after running `$> make 1.0-core-update-api` (Core) / `$> make 1.0-ui-update-api` (UI) if there are changes / errors we're 🆗 with (e.g. `AddedMethod` changes are marked as errors but don't break SemVer) 🚀 If there are SemVer breaking changes add the `SEMVER` label. See [Metalava](https://github.com/mapbox/mapbox-navigation-android/blob/master/docs/metalava.md) docs

### CODE REVIEW

**Comment by kmadsen** [comment]
cc: @LukasPaczos @Guardiola31337

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/TripSession.kt [hunk 1]
```diff
@@ -23,6 +20,7 @@ internal interface TripSession {
 
     fun start()
     fun stop()
+    fun refresh(locationEngineRequest: LocationEngineRequest)
```
Comment (LukasPaczos) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/TripSession.kt:23]: Is there a use-case for changing the request in runtime? If there's no clear one, I'd rather remove it.

Comment (kmadsen) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/TripSession.kt]: Yeah am also fine to remove, we discussed offline reasons to support it such as; customer wants to lower the request to save battery. Which is something you would do at runtime when battery changes.

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/trip/session/MapboxTripSession.kt [hunk 2]
```diff
@@ -94,6 +93,9 @@ class MapboxTripSession(
     private val bannerInstructionEvent = BannerInstructionEvent()
     private val voiceInstructionEvent = VoiceInstructionEvent()
 
+    private val locationEngineRequest: LocationEngineRequest = LocationEngineRequest.Builder(1000L)
+        .setPriority(LocationEngineRequest.PRIORITY_HIGH_ACCURACY)
```
Comment (LukasPaczos) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/trip/session/MapboxTripSession.kt:97]: Let's also set the fastest interval to something reasonable, like 500ms.

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt [hunk 3]
```diff
@@ -120,19 +118,11 @@ private const val MAPBOX_NAVIGATION_TOKEN_EXCEPTION = "You need to provide an ac
  *
  * @param navigationOptions a set of [NavigationOptions] used to customize various features of the SDK.
  * Use [defaultNavigationOptionsBuilder] to set default options
- * @param locationEngine used to listen for raw location updates
- * @param locationEngineRequest used to request raw location updates
  */
 class MapboxNavigation
-@JvmOverloads
 constructor(
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt:123]: We can remove `constructor`

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/trip/session/MapboxTripSession.kt [hunk 4]
```diff
@@ -72,6 +70,16 @@ class MapboxTripSession(
 
     companion object {
         private const val STATUS_POLLING_INTERVAL = 1000L
+
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/trip/session/MapboxTripSession.kt:73]: NIT unnecessary new line

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/trip/session/MapboxTripSession.kt [hunk 5]
```diff
@@ -72,6 +70,16 @@ class MapboxTripSession(
...
+        private const val LOCATION_POLLING_INTERVAL = 1000L
+
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/trip/session/MapboxTripSession.kt:75]: NIT unnecessary new line

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt (hunk 3 above)
Comment (kmadsen) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt:123]: https://github.com/mapbox/mapbox-navigation-android/pull/3253

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/trip/session/MapboxTripSession.kt (hunk 4 above)
Comment (kmadsen) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/trip/session/MapboxTripSession.kt:73]: Following up here https://github.com/mapbox/mapbox-navigation-android/pull/3253

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/trip/session/MapboxTripSession.kt (hunk 5 above)
Comment (kmadsen) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/trip/session/MapboxTripSession.kt:75]: Following up here https://github.com/mapbox/mapbox-navigation-android/pull/3253

**Pull request merged**